src/test/java/unicash/ui/RollingBalanceUiTest.java

```diff
@@ -0,0 +1,179 @@
+package unicash.ui;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static unicash.model.util.SampleDataUtil.getSampleUniCash;
+import static unicash.testutil.TestDataUtil.getSumOfTestExpenses;
+import static unicash.testutil.TestDataUtil.getSumOfTestIncomes;
+import static unicash.testutil.TestDataUtil.getTestTransactions;
+import static unicash.testutil.TestDataUtil.getTestTransactionsAsUserInputs;
+
+import java.util.Collections;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.testfx.api.FxRobot;
+import org.testfx.api.FxToolkit;
+import org.testfx.framework.junit5.ApplicationExtension;
+import org.testfx.util.WaitForAsyncUtils;
+
+import javafx.scene.control.Label;
+import javafx.scene.input.KeyCode;
+import unicash.MainApp;
+import unicash.commons.enums.CommandType;
+import unicash.commons.enums.TransactionType;
+import unicash.model.transaction.Transaction;
+import unicash.model.transaction.predicates.TransactionNameContainsKeywordsPredicate;
+import unicash.testutil.TransactionBuilder;
+import unicash.testutil.UserInputBuilder;
+
+@ExtendWith(ApplicationExtension.class)
+public class RollingBalanceUiTest {
+
+    @BeforeEach
+    public void setUp() throws TimeoutException {
+        FxToolkit.registerPrimaryStage();
+        FxToolkit.setupApplication(() -> new MainApp());
```

Comment:
Please use the `MainApp(Path)` variant instead. Using just `MainApp()` will cause your unit tests to override the actual data file on your machine and might cause problems with the CI if that data is needed. See this for reference: https://ay2324s1-cs2103-t16-3.github.io/tp/DeveloperGuide.html#using-beforeeach

Reply:
Thanks for letting me know! I have added the change.